lib/std/os/linux/s390x.zig

```diff
@@ -76,6 +76,8 @@ pub fn syscall5(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize,
 }
 
 pub fn syscall6(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize, arg6: usize) usize {
+    if (number == .mmap) return syscall1(number, @intFromPtr(&[_]usize{ arg1, arg2, arg3, arg4, arg5, arg6 }));
```

Comment:
hang on, let's discuss this... why is this here and not somewhere else?

Reply:
You mean as opposed to moving it up to the `mmap()` wrapper? That's an option, but it would mean we'd also have to duplicate the special-casing here:

https://github.com/ziglang/zig/blob/9b273f6b9acaf288920e8320adaa34f30348e97f/lib/std/os/linux/tls.zig#L532-L554

I don't have a strong preference about how we do it; I just followed musl's lead by putting it here.

For what it's worth, this only affects `mmap()` on i386 and s390(x) for historical reasons; these system calls existed before the Linux syscall mechanism gained support for 5+ parameters.

Reply:
I have a strong preference for it being there - it has to do with mmap, not to do with syscall5. This also makes it a compile-time branch instead of a runtime branch.

Reply:
Ok, moved it to the syscall wrappers.